client/shipping-label/views/purchase/steps/rates/list.js

```diff
@@ -63,7 +61,7 @@ const ShippingRates = ( {
 
 	return (
 		<div>
-			{ renderRateNotice( showRateNotice ) }
+			{ shouldShowRateNotice && renderRateNotice() }
```

Comment:
Just to make the code clearer, you could go with something like:
`shouldShowRateNotice ? renderRateNotice() : ''`

PS: the previous implementation of `renderRateNotice()` returning a null wasn't much better. It should have also returned an empty string if the boolean was false.